File: server/app/models/construction.py
from sqlalchemy import Column, Integer, String, Text, Boolean, Float, DateTime
from sqlalchemy.orm import declarative_base
from datetime import datetime

Base = declarative_base()

class Construction(Base):
    __tablename__ = "constructions"

    id = Column(Integer, primary_key=True, index=True)
    place = Column(String, index=True)
    period = Column(String)
    description = Column(Text)
    is_published = Column(Boolean, default=False)
    thumbnail = Column(String, nullable=True)
    total_price = Column(Float, nullable=True)
    created_at = Column(DateTime, default=datetime.now)

File: client/types/construction.ts
export interface ConstructionResponse {
  id: number;
  place: string;
  period: string;
  description: string;
  is_published: boolean;
  thumbnail?: string;
  created_at: string;
  total_price?: number;
}

export interface ConstructionCreateRequest {
  place: string;
  period: string;
  description: string;
  is_published: boolean;
  thumbnail?: string;
  total_price?: number;
}
